### PR TITLE
Enable alternative reading of a registry module by its ID

### PR DESCRIPTION
**Pull request by dsa0x**
## Description

This enables an alternative way to read a registry module by its ID instead of its address (namespace/name/system). When the external ID is given, all other attributes are ignored.

## Testing plan

- Create a token in your HCP TF instance
- Add the TFE_ADDRESS and TFE_TOKEN variables to your environment
- Running `go test -v -run ^TestRegistryModulesRead$ github.com/hashicorp/go-tfe` should succeed.

## External links

- [JIRA](https://hashicorp.atlassian.net/browse/TFECO-6399)

## Output from tests
Including output from tests may require access to a TFE instance. Ignore this section if you have no environment to test against.

```
$ TFE_ADDRESS="https://example" TFE_TOKEN="example" go test -v -run ^TestRegistryModulesRead$ github.com/hashicorp/go-tfe
=== RUN   TestRegistryModulesRead/with_a_external_ID_field_for_private_module
=== RUN   TestRegistryModulesRead/with_a_external_ID_field_for_private_module/permissions_are_properly_decoded
=== RUN   TestRegistryModulesRead/with_a_external_ID_field_for_private_module/timestamps_are_properly_decoded
...
    --- PASS: TestRegistryModulesRead/with_a_external_ID_field_for_private_module (0.21s)
        --- PASS: TestRegistryModulesRead/with_a_external_ID_field_for_private_module/permissions_are_properly_decoded (0.00s)
        --- PASS: TestRegistryModulesRead/with_a_external_ID_field_for_private_module/timestamps_are_properly_decoded (0.00s)
...
PASS
ok      github.com/hashicorp/go-tfe     3.926s

...
```

### CODE REVIEW

File: tfe.go
Comment (dsa0x) [inline tfe.go:322]: This fixes a bug where the original set query param is overwritten.

File: tfe.go
Comment (dsa0x) [inline tfe.go:327]: This fixes a bug where the original set query param is overwritten.

File: registry_module.go
Comment (brandonc) [inline registry_module.go:532]: I know you didn't add this, but registry_module.go seems to be the only file in go-tfe which is using the `log`  package, and, as a library, probably shouldn't be logging to stderr. It would be nice if you could go ahead and remove all these log.Print calls from this file, but I'll take care of it if you don't feel like waiting for another review!

**Review by brandonc — APPROVED** [review]

File: registry_module.go
Comment (dsa0x) [reply registry_module.go:532]: Thank you. I will go ahead and merge this. Is there a convention employed in go-tfe for including deprecated messages?. Perhaps we could just remove them entirely, and have the comments indicate the deprecated attributes.

**Pull request merged**
